### PR TITLE
Add xml comments to ExecutionStrategy and ExecutionNode

### PR DESCRIPTION
**Pull request by Shane32**

### CODE REVIEW

**Comment by sungam3r** [comment]
I will review this tomorrow.

File: src/GraphQL/Execution/ExecutionNode.cs [hunk 1]
```diff
@@ -1,23 +1,57 @@
...
+    /// </summary>
     public abstract class ExecutionNode
     {
+        /// <summary>
+        /// Returns the parent node, or null if this is the root node.
+        /// </summary>
         public ExecutionNode Parent { get; }
+
+        /// <summary>
+        /// Returns the underlying graph type of this node (does not represent a <see cref="NonNullGraphType"/> or <see cref="ListGraphType"/> instance).
```
Comment (sungam3r) [inline src/GraphQL/Execution/ExecutionNode.cs:20]: As I can see from https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Execution/ExecutionStrategy.cs#L154 it is not true.

File: src/GraphQL/Execution/ExecutionNode.cs [hunk 2]
```diff
@@ -132,20 +178,40 @@ private IEnumerable<object> GeneratePath(bool preferAlias)
...
+    /// </summary>
     public class ObjectExecutionNode : ExecutionNode, IParentExecutionNode
     {
+        /// <summary>
+        /// Returns a dictionary of child execution nodes, with keys set to the names of the child fields that the child nodes represent.
+        /// </summary>
         public IDictionary<string, ExecutionNode> SubFields { get; set; }
 
+        /// <summary>
+        /// Initializes an instances of <see cref="ObjectExecutionNode"/> with the specified values.
```
Comment (sungam3r) [inline src/GraphQL/Execution/ExecutionNode.cs:203]:
```suggestion
        /// Initializes an instance of <see cref="ObjectExecutionNode"/> with the specified values.
```

File: src/GraphQL/Execution/ExecutionStrategy.cs [hunk 3]
```diff
@@ -205,7 +236,8 @@ protected virtual async Task ExecuteNodeAsync(ExecutionContext context, Executio
         }
 
         /// <summary>
-        /// Completes a pending data loader node. If the node does not return a IDataLoaderResult, it will build child nodes, but does not execute them.
+        /// Completes a pending data loader node.  If the node does not return an <see cref="IDataLoaderResult"/>,
```
Comment (sungam3r) [inline src/GraphQL/Execution/ExecutionStrategy.cs:239]:
```suggestion
        /// Completes a pending data loader node. If the node does not return an <see cref="IDataLoaderResult"/>,
```

File: src/GraphQL/Execution/ExecutionNode.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQL/Execution/ExecutionNode.cs:20]: Correct; I will fix it.

Comment (Shane32) [reply src/GraphQL/Execution/ExecutionNode.cs:20]:
```suggestion
        /// Returns the underlying graph type of this node; does not represent a <see cref="NonNullGraphType"/>, and
        /// for child nodes of an array execution node, does not represent a <see cref="ListGraphType"/> instance.
```

Comment (Shane32) [reply src/GraphQL/Execution/ExecutionNode.cs]: Please review @sungam3r

**Review by sungam3r — APPROVED** [review]

**Pull request merged**